sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,470 @@
+     * "application/json" type. Note that "application/json" is still a possible value for this field.
+     * @return the content type the data is in, or null if the data is nonexistent or in "application/json" format.
+     */
+    public String getDataContentType() {
+        return this.dataContentType;
+    }
+
+    /**
+     * Get the schema that the data adheres to.
+     * @return a URI of the data schema, or null if it is not set.
+     */
+    public String getDataSchema() {
+        return this.dataSchema;
+    }
+
+    /**
+     * Set the schema that the data adheres to.
+     * @param dataSchema a URI identifying the schema of the data.
+     *
+     * @return the cloud event itself.
+     */
+    public CloudEvent setDataSchema(String dataSchema) {
+        this.dataSchema = dataSchema;
+        return this;
+    }
+
+    /**
+     * Get the subject associated with this event.
+     * @return the subject, or null if the subject was not set.
+     */
+    public String getSubject() {
+        return this.subject;
+    }
+
+    /**
+     * Set the subject of the event.
+     * @param subject the subject to set.
+     *
+     * @return the cloud event itself.
+     */
+    public CloudEvent setSubject(String subject) {
+        this.subject = subject;
+        return this;
+    }
+
+    /**
+     * Get a map of the additional user-defined attributes associated with this event.
+     * @return the extension attributes as an unmodifiable map.
+     */
+    @JsonAnyGetter
+    public Map<String, Object> getExtensionAttributes() {
+        return this.extensionAttributes;
+    }
+
+    /**
+     * Add/Overwrite a single extension attribute to the cloud event.
+     * @param name the name of the attribute. It must contains only alphanumeric characters and not be be any
+     *             CloudEvent reserved attribute names.
+     * @param value the value to associate with the name.
+     *
+     * @return the cloud event itself.
+     * @throws IllegalArgumentException if name format isn't correct.
+     */
+    @JsonAnySetter
+    public CloudEvent addExtensionAttribute(String name, Object value) {
+        if (Objects.isNull(name)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'name' cannot be null."));
+        }
+        if (Objects.isNull(value)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'value' cannot be null."));
+        }
+        if (!validateAttributeName(name)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                "'name' must have only small-case alphanumeric characters and not be one of the CloudEvent reserved "
+                    + "attribute names"));
+        }
+        if (this.extensionAttributes == null) {
+            this.extensionAttributes = new HashMap<>();
+        }
+        this.extensionAttributes.put(name.toLowerCase(Locale.ENGLISH), value);
```

Comment:
nit: shouldn't need the toLowerCase here.

Reply:
Good catch. This has no use since we checked whether it's lower case.